crates/qasm3/src/circuit.rs

```diff
@@ -264,24 +270,36 @@ impl PyCircuitModule {
         self.clbit.call0(py)
     }
 
-    pub fn new_instruction<O, Q, C>(
-        &self,
-        py: Python,
+    pub fn new_instruction<'a, O, Q, C>(
+        &'a self,
+        py: Python<'a>,
         operation: O,
         qubits: Q,
         clbits: C,
     ) -> PyResult<Py<PyAny>>
     where
-        O: IntoPy<Py<PyAny>>,
-        Q: IntoPy<Py<PyTuple>>,
-        C: IntoPy<Py<PyTuple>>,
+        O: IntoPyObject<'a>,
+        Q: IntoPyObject<'a>,
+        C: IntoPyObject<'a>,
+        <Q as pyo3::IntoPyObject<'a>>::Output: pyo3::IntoPyObject<'a>,
+        <C as pyo3::IntoPyObject<'a>>::Output: pyo3::IntoPyObject<'a>,
+        pyo3::PyErr: From<<Q as pyo3::IntoPyObject<'a>>::Error>,
+        pyo3::PyErr: From<<C as pyo3::IntoPyObject<'a>>::Error>,
     {
-        self.circuit_instruction
-            .call1(py, (operation, qubits.into_py(py), clbits.into_py(py)))
+        self.circuit_instruction.call1(
+            py,
+            (
+                operation,
+                qubits.into_pyobject(py)?,
+                clbits.into_pyobject(py)?,
+            ),
+        )
```

Comment:
I think this could be `IntoPyObjectExt::into_pyobject_or_pyerr`, which would automatically use the trait bounds from the `IntoPyObject::Error` associated type to avoid needing to spell them out again.

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/13577/commits/0577f48a7814455551ec2cda3540a9d1a7ad9847